app/mobile-content/shared/star-rating/star-rating.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, Input, Output, EventEmitter, OnChanges, SimpleChanges } from '@angular/core';
+
+@Component({
+  moduleId: module.id,
+  selector: 'hip-star-rating',
+  templateUrl: 'star-rating.component.html',
+  styleUrls: ['star-rating.component.css']
+})
+export class RatingComponent implements OnChanges {
+  private range: {rating: number, iconValue: string }[] = [];
+  private _value;
+
+  @Input() rating: number;
+  @Input() exhibitId: number;
+  @Output() ratingClicked: EventEmitter<any> = new EventEmitter<any>();
+
+  getStarRating() {
```

Comment:
This looks awfully complicated. At least explain in some comments, what each step does.